// @codeday/topo/Atom/Skelly/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _core = require("@emotion/core");

var _Box = _interopRequireDefault(require("../Box"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  animation: ", " 8s ease-in-out infinite;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  from {\n    background-position: 200% 0;\n  }\n  to {\n    background-position: -200% 0;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var load = (0, _core.keyframes)(_templateObject());
var loadAnimation = (0, _core.css)(_templateObject2(), load);

var Skelly = function Skelly(props) {
  return (0, _core.jsx)(_Box["default"], _extends({
    grad: "skelly",
    height: "1em",
    backgroundSize: "800% 100%",
    marginBottom: 1,
    borderRadius: "sm",
    css: [loadAnimation]
  }, props));
};

Skelly.displayName = 'Skelly';
var _default = Skelly;
exports["default"] = _default;
