// @codeday/topo/Organism/Header/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "SiteLogo", {
  enumerable: true,
  get: function get() {
    return _siteLogo["default"];
  }
});
Object.defineProperty(exports, "Menu", {
  enumerable: true,
  get: function get() {
    return _menu["default"];
  }
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Box = _interopRequireDefault(require("../../Atom/Box"));

var _Content = _interopRequireDefault(require("../../Molecule/Content"));

var _utils = require("../../_utils");

var _siteLogo = _interopRequireDefault(require("./site-logo"));

var _menu = _interopRequireDefault(require("./menu"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Header = function Header(_ref) {
  var darkBackground = _ref.darkBackground,
      underscore = _ref.underscore,
      children = _ref.children,
      gradAmount = _ref.gradAmount,
      props = _objectWithoutProperties(_ref, ["darkBackground", "underscore", "children", "gradAmount"]);

  var childrenWithProps = _react["default"].Children.map(children, (0, _utils.setChildProps)({
    darkBackground: darkBackground
  }));

  var logo = (0, _utils.childrenOfType)(childrenWithProps, _siteLogo["default"]);
  var menu = (0, _utils.childrenOfType)(childrenWithProps, _menu["default"]);
  return /*#__PURE__*/_react["default"].createElement(_Box["default"], _extends({
    grad: (darkBackground && gradAmount !== false || gradAmount) && "".concat(darkBackground ? 'darken' : 'lighten', ".").concat(gradAmount || 'sm', ".180")
  }, props), /*#__PURE__*/_react["default"].createElement(_Content["default"], {
    padding: 3,
    paddingTop: 6,
    paddingBottom: 4
  }, /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    marginBottom: 6,
    paddingBottom: 4,
    borderBottomColor: darkBackground ? 'whiteAlpha.300' : 'gray.200',
    borderBottomWidth: underscore ? '1px' : 0
  }, /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    "float": "left"
  }, logo), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    "float": "right",
    textAlign: "right"
  }, menu), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    style: {
      clear: 'both'
    }
  }))));
};

Header.propTypes = {
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node]).isRequired,
  darkBackground: _propTypes["default"].bool,
  underscore: _propTypes["default"].bool,
  gradAmount: _propTypes["default"].string
};
Header.defaultProps = {
  darkBackground: false,
  underscore: false,
  gradAmount: null
};
var _default = Header;
exports["default"] = _default;
