// @codeday/topo/Organism/Footer/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.CustomText = exports.CustomLinks = void 0;

var _react = _interopRequireWildcard(require("react"));

var _List = _interopRequireWildcard(require("../../Atom/List"));

var _Box = _interopRequireWildcard(require("../../Atom/Box"));

var _Skelly = _interopRequireDefault(require("../../Atom/Skelly"));

var _Text = _interopRequireWildcard(require("../../Atom/Text"));

var _Content = _interopRequireDefault(require("../../Molecule/Content"));

var _utils = require("../../_utils");

var _utils2 = require("../../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(n); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return; var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var CustomLinks = (0, _utils.makePureBox)('Custom Links');
exports.CustomLinks = CustomLinks;
var CustomText = (0, _utils.makePureBox)('CustomText');
exports.CustomText = CustomText;

var StandardLinks = function StandardLinks() {
  var _useTheme = (0, _utils2.useTheme)(),
      contentful = _useTheme.contentful;

  var _useState = (0, _react.useState)(null),
      _useState2 = _slicedToArray(_useState, 2),
      links = _useState2[0],
      setLinks = _useState2[1];

  (0, _react.useEffect)(function () {
    if (!contentful) return;
    var base = "https://cdn.contentful.com/spaces/".concat(contentful.spaceId, "/environments/master");
    var endpoint = "/entries?content_type=site&access_token=".concat(contentful.token);

    _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee() {
      var json;
      return regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              _context.next = 2;
              return fetch("".concat(base).concat(endpoint));

            case 2:
              _context.next = 4;
              return _context.sent.json();

            case 4:
              json = _context.sent;
              setLinks(json.items.map(function (i) {
                return i.fields;
              }).filter(function (i) {
                return i.type === 'Public' && i.display.includes('Footer');
              }));

            case 6:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }))();
  }, [contentful]);
  return /*#__PURE__*/_react["default"].createElement(_List["default"], null, !links ? /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_List.Item, null, /*#__PURE__*/_react["default"].createElement(_Skelly["default"], null)), /*#__PURE__*/_react["default"].createElement(_List.Item, null, /*#__PURE__*/_react["default"].createElement(_Skelly["default"], null)), /*#__PURE__*/_react["default"].createElement(_List.Item, null, /*#__PURE__*/_react["default"].createElement(_Skelly["default"], null))) : links.map(function (_ref2) {
    var title = _ref2.title,
        link = _ref2.link;
    return /*#__PURE__*/_react["default"].createElement(_List.Item, null, /*#__PURE__*/_react["default"].createElement(_Text.Link, {
      href: link,
      target: "_blank",
      key: link
    }, title));
  }));
};

var Footer = (0, _utils.pureRef)(function (_ref3, ref) {
  var children = _ref3.children;
  var customLinks = (0, _utils.childrenOfType)(children, CustomLinks);
  var customText = (0, _utils.childrenOfType)(children, CustomText);
  return /*#__PURE__*/_react["default"].createElement(_Content["default"], {
    ref: ref
  }, /*#__PURE__*/_react["default"].createElement(_Box.Grid, {
    templateColumns: {
      base: '1fr',
      md: '6fr 3fr 3fr'
    },
    color: "current.textLight"
  }, /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    gridRow: {
      base: 3,
      md: 1
    },
    marginTop: {
      base: 6,
      md: 0
    }
  }, /*#__PURE__*/_react["default"].createElement(_Box["default"], null, customText.length > 0 ? customText : /*#__PURE__*/_react["default"].createElement(_Text["default"], null, "Copyright \xA9 2009 \u2013 ", new Date().getFullYear(), " CodeDay.", /*#__PURE__*/_react["default"].createElement("br", null), "A 501(c)(3) nonprofit. EIN: ", /*#__PURE__*/_react["default"].createElement(_Text.CopyText, null, "26-4742589"), ".", /*#__PURE__*/_react["default"].createElement("br", null), /*#__PURE__*/_react["default"].createElement(_Text.Link, {
    href: "tel:18886077763"
  }, "(888) 607-7763"))), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    marginTop: 4
  }, /*#__PURE__*/_react["default"].createElement(_Text.Link, {
    href: "https://www.srnd.org/privacy"
  }, "Privacy & Cookies"), /*#__PURE__*/_react["default"].createElement("br", null), /*#__PURE__*/_react["default"].createElement(_Text.Link, {
    href: "https://www.srnd.org/privacy/controls",
    target: "_blank"
  }, "Do Not Sell My Personal Information"))), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    gridRow: {
      base: 2,
      md: 1
    },
    marginTop: {
      base: customLinks.length > 0 && 6,
      md: 0
    }
  }, customLinks.length > 0 && /*#__PURE__*/_react["default"].createElement(_Text.Heading, {
    as: "h2",
    fontSize: "xl"
  }, "More"), customLinks), /*#__PURE__*/_react["default"].createElement(_Box["default"], null, /*#__PURE__*/_react["default"].createElement(_Text.Heading, {
    as: "h2",
    fontSize: "xl"
  }, "Resources"), /*#__PURE__*/_react["default"].createElement(StandardLinks, null))));
});
Footer.propTypes = _Box["default"].propTypes;
Footer.defaultProps = _Box["default"].defaultProps;
var _default = Footer;
exports["default"] = _default;
