// @codeday/topo/Atom/StatusDot/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Box = _interopRequireDefault(require("../Box"));

var _recompose = require("recompose");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var StatusDot = (0, _recompose.withProps)(function (props) {
  return {
    width: props.size || 3,
    height: props.size || 3,
    display: 'inline-block',
    bg: props.online && 'green.500' || (props.away || props.pending) && 'orange.400' || 'red.600',
    borderRadius: 'full',
    title: props.online && 'online' || props.away && 'away' || props.pending && 'pending' || 'offline'
  };
})(_Box["default"]);
StatusDot.displayName = 'StatusDot';
StatusDot.propTypes = {
  size: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].string]),
  online: _propTypes["default"].bool,
  away: _propTypes["default"].bool,
  pending: _propTypes["default"].bool,
  offline: _propTypes["default"].bool
};
StatusDot.defaultProps = {
  size: null,
  online: false,
  away: false,
  pending: false,
  offline: false
};
var _default = StatusDot;
exports["default"] = _default;
