// @codeday/topo/Molecule/Deck/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactPdf = require("react-pdf");

var _UiArrowLeft = _interopRequireDefault(require("@codeday/topocons/Icon/UiArrowLeft"));

var _UiArrowRight = _interopRequireDefault(require("@codeday/topocons/Icon/UiArrowRight"));

var _UiDownload = _interopRequireDefault(require("@codeday/topocons/Icon/UiDownload"));

var _Box = _interopRequireWildcard(require("../../Atom/Box"));

var _Button = _interopRequireDefault(require("../../Atom/Button"));

var _Spinner = _interopRequireDefault(require("../../Atom/Spinner"));

var _utils = require("../../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && Symbol.iterator in Object(iter)) return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(n); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return; var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

_reactPdf.pdfjs.GlobalWorkerOptions.workerSrc = "//cdnjs.cloudflare.com/ajax/libs/pdf.js/".concat(_reactPdf.pdfjs.version, "/pdf.worker.js");
var Deck = (0, _react.forwardRef)(function (_ref, ref) {
  var src = _ref.src,
      allowDownload = _ref.allowDownload;

  var _useState = (0, _react.useState)(300),
      _useState2 = _slicedToArray(_useState, 2),
      width = _useState2[0],
      setWidth = _useState2[1];

  var _useState3 = (0, _react.useState)(0),
      _useState4 = _slicedToArray(_useState3, 2),
      height = _useState4[0],
      setHeight = _useState4[1];

  var _useState5 = (0, _react.useState)(1),
      _useState6 = _slicedToArray(_useState5, 2),
      pageNumber = _useState6[0],
      setPageNumber = _useState6[1];

  var _useState7 = (0, _react.useState)(null),
      _useState8 = _slicedToArray(_useState7, 2),
      totalPages = _useState8[0],
      setTotalPages = _useState8[1];

  var myRef = ref || (0, _react.useRef)();
  var theme = (0, _utils.useTheme)();
  var loading = (0, _react.useMemo)(function () {
    return /*#__PURE__*/_react["default"].createElement(_Box["default"], {
      textAlign: "center"
    }, /*#__PURE__*/_react["default"].createElement(_Spinner["default"], null));
  }, []);

  var navigate = function navigate(amount) {
    return setPageNumber(Math.min(totalPages, Math.max(1, amount + pageNumber)));
  };

  var filter = 'drop-shadow( 0 0 1px rgba(0, 0, 0, .7))'; // This will throw an error server-side.

  if (typeof window === 'undefined') return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null);
  return /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    position: "relative"
  }, /*#__PURE__*/_react["default"].createElement(_Box.SizeBox, {
    onWidthChanged: setWidth,
    borderWidth: "1px",
    borderColor: "gray.100",
    borderRadius: allowDownload ? "".concat(theme.radii.md, " ").concat(theme.radii.md, " 0 ").concat(theme.radii.md) : 'md',
    "aria-hidden": true
  }, /*#__PURE__*/_react["default"].createElement(_reactPdf.Document, {
    file: src,
    inputRef: myRef,
    loading: loading,
    onLoadSuccess: function onLoadSuccess(_ref2) {
      var numPages = _ref2.numPages;
      return setTotalPages(numPages);
    },
    onClick: function onClick() {
      return setPageNumber(pageNumber + 1);
    }
  }, totalPages && _toConsumableArray(Array(totalPages).keys()).map(function (number) {
    return /*#__PURE__*/_react["default"].createElement(_Box["default"], {
      display: number + 1 === pageNumber ? 'block' : 'none',
      "aria-hidden": number + 1 !== pageNumber
    }, /*#__PURE__*/_react["default"].createElement(_reactPdf.Page, {
      loading: loading,
      width: width - 2,
      pageNumber: number + 1,
      onRenderSuccess: function onRenderSuccess() {
        return setHeight(myRef.current.offsetHeight);
      }
    }));
  }))), totalPages && /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    height: "".concat(height, "px"),
    width: "".concat(width, "px"),
    display: "block",
    cursor: "pointer",
    position: "absolute",
    zIndex: "1000",
    top: "0",
    marginBottom: 0
  }, /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    width: "40%",
    height: "100%",
    display: "inline-block",
    textAlign: "right",
    color: "white",
    position: "relative",
    onClick: function onClick() {
      return navigate(-1);
    }
  }, pageNumber > 1 && /*#__PURE__*/_react["default"].createElement(_UiArrowLeft["default"], {
    "aria-placeholder": "Previous Slide",
    style: {
      position: 'absolute',
      top: '50%',
      left: '0.5rem',
      filter: filter,
      fill: 'white'
    }
  })), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    width: "60%",
    height: "100%",
    display: "inline-block",
    color: "white",
    onClick: function onClick() {
      return navigate(1);
    }
  }, pageNumber < totalPages && /*#__PURE__*/_react["default"].createElement(_UiArrowRight["default"], {
    "aria-placeholder": "Next Slide",
    style: {
      position: 'absolute',
      top: '50%',
      right: '0.5rem',
      filter: filter,
      fill: 'white'
    }
  }))), /*#__PURE__*/_react["default"].createElement(_Box["default"], {
    textAlign: "right",
    style: allowDownload ? {} : {
      position: 'absolute',
      left: '-99999px'
      /* needed for screen readers */

    }
  }, /*#__PURE__*/_react["default"].createElement(_Button["default"], {
    size: "sm",
    variant: "outline",
    as: "a",
    href: src,
    target: "_blank",
    borderRadius: "0 0 ".concat(theme.radii.md, " ").concat(theme.radii.md),
    borderTop: "none",
    color: "gray.600"
  }, /*#__PURE__*/_react["default"].createElement(_UiDownload["default"], null), "\xA0Download PDF")));
});
Deck.propTypes = {
  src: _propTypes["default"].string.isRequired,
  allowDownload: _propTypes["default"].bool
};
Deck.defaultProps = {
  allowDownload: false
};
var _default = Deck;
exports["default"] = _default;
